src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -67,121 +94,489 @@ public Future<String> handle(KafkaConsumerRecord<String, String> event) {
       Map<String, String> kafkaHeaders = KafkaHeaderUtils.kafkaHeadersToMap(event.headers());
       String instanceId = sharingInstanceMetadata.getInstanceIdentifier().toString();
 
-      LOGGER.info("Event CONSORTIUM_INSTANCE_SHARING_INIT has been received for instanceId: {}, sourceTenant: {}, targetTenant: {}",
+      LOGGER.info("Event CONSORTIUM_INSTANCE_SHARING_INIT has been received for InstanceId={}, sourceTenant={}, targetTenant={}",
         instanceId, sharingInstanceMetadata.getSourceTenantId(), sharingInstanceMetadata.getTargetTenantId());
 
-      Context targetTenantContext = EventHandlingUtil.constructContext(sharingInstanceMetadata.getTargetTenantId(),
-        kafkaHeaders.get(TOKEN.toLowerCase()), kafkaHeaders.get(URL.toLowerCase()));
-      InstanceCollection targetInstanceCollection = storage.getInstanceCollection(targetTenantContext);
+      InstanceCollection targetInstanceCollection =
+        getTenantSpecificSourceCollection(sharingInstanceMetadata.getTargetTenantId(), kafkaHeaders);
 
-      Context sourceTenantContext = EventHandlingUtil.constructContext(sharingInstanceMetadata.getSourceTenantId(),
-        kafkaHeaders.get(TOKEN.toLowerCase()), kafkaHeaders.get(URL.toLowerCase()));
-      InstanceCollection sourceInstanceCollection = storage.getInstanceCollection(sourceTenantContext);
+      InstanceCollection sourceInstanceCollection =
+        getTenantSpecificSourceCollection(sharingInstanceMetadata.getSourceTenantId(), kafkaHeaders);
 
-      return checkIsInstanceExistsOnTargetTenant(sharingInstanceMetadata, targetInstanceCollection,
-        sourceInstanceCollection, kafkaHeaders);
+      return checkIsInstanceExistsOnTargetTenant(sharingInstanceMetadata, targetInstanceCollection, sourceInstanceCollection, kafkaHeaders);
     } catch (Exception ex) {
       LOGGER.error(format("Failed to process data import kafka record from topic %s", event.topic()), ex);
       return Future.failedFuture(ex);
     }
   }
 
   private Future<String> checkIsInstanceExistsOnTargetTenant(SharingInstance sharingInstanceMetadata,
```

Comment:
Change the name of the method to show that an instance is published as well. Not just checking for instance existence